src/components/TextField/TextField.tsx

```diff
@@ -56,94 +24,65 @@ export type Props = {
   onKeyDown?: React.KeyboardEventHandler<HTMLTextAreaElement | HTMLInputElement>;
   /** Callback fired when the `input` value typed is changed */
   onInput?: React.EventHandler<any>;
-};
+} & TextInputWrapperProps;
```

Comment:
As discussed this is not in the scope of the TextArea component but should be resolved.

Consider changing the type of the `leftIcon` and `rightIcon` from  `string | JSX.Element | null;` to `JSX.Element | null`